File: kt/src/main/kotlin/com/github/scglwsj/parking/ParkingBoy.kt
package com.github.scglwsj.parking

class ParkingBoy(val parkingLots: List<ParkingLot>) {
    fun park(car: Car): Ticket {
        for (lot in parkingLots) {
            try {
                return lot.park(car)
            } catch (e: Exception) {
            }
        }

        throw Exception("capacity is not enough.")
    }
}

File: cs/Parking/ParkingBoy.cs
using System;
using System.Collections.Generic;

namespace Parking
{
    public class ParkingBoy
    {
        public IReadOnlyList<ParkingLot> ParkingLots;

        public ParkingBoy(IReadOnlyList<ParkingLot> parkingLots)
        {
            ParkingLots = parkingLots;
        }

        public Ticket Park(Car car)
        {
            foreach (var parkingLot in ParkingLots)
            {
                try
                {
                    return parkingLot.Park(car);
                }
                catch { }
            }
            throw new Exception("capacity is not enough.");
        }
    }
}
